onnxruntime/core/graph/matmul_add_fusion.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/initializer.h"
+#include "core/graph/matmul_add_fusion.h"
+#include "core/graph/graph_utils.h"
+
+using namespace onnx;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+Status MatMulAddFusion::Apply(Graph& graph, bool& modified) const {
+  GraphViewer graph_viewer(graph);
+  const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
+  std::vector<onnxruntime::NodeIndex> removed_nodes;
+
+  for (auto node_index : node_topology_list) {
```

Comment:
It is not necessary to transverse by top logical order.

Reply:
As discussed offline, if top logical order is used, then you can move graph.RemoveNode(i); in the loop to reduce a few line of code.

Reply:
Still need the top pological order, and remove node in reversed order out of the loop.